pkg/rpc/server/server.go

```diff
@@ -86,6 +89,14 @@ type (
 		transactionCh     chan *transaction.Transaction
 		notaryRequestCh   chan mempoolevent.Event
 	}
+
+	// session holds a set of iterators got after invoke* call with corresponding
+	// finalizer and session expiration time.
+	session struct {
+		Iterators map[string]stackitem.Item // TODO: convert it to array, we don't need map here. Most of the time it's holding a single iterator per session.
```

Comment:
@fyrchik, what do you think, is it better to store iterators as an array or as a map?

Reply:
I vote for slice, because I think the common usecase will not involve a lot of iterators, where using map would benefit us.

Reply:
Fixed. I've also added a mutex inside sessions so that iterators are protected by this mutex.